### DISCUSSION TITLE
Should we turn on ASP .NET Diagnostics extension by default for web apps?

### DISCUSSION THREAD

**Discussion by isaacabraham** [CompositionalIT/farmer#491] [Q&A]
This would turn on the App Service extension (which is I believe the same as [this](https://www.nuget.org/packages/Microsoft.Extensions.Logging.AzureAppServices/) nuget package) if the App Service was netcore.

What do you think? I've seen enough people struggle with the log stream in the past to think it could be a good idea.

@Dzoukr @davedawkins

**Comment by davedawkins** [comment]
Sure, on by default and have -no_logging to switch it off? It should be
easy to init the extensions list with the log ext

**Reply by isaacabraham** [reply]
@davedawkins there would be a flat to disable the logging by default, yes.

**Comment by Dzoukr** [comment]
That's always a difficult thing to make sane default. If it doesn't have any impact on pricing on Azure, let's have it as default. If it has, I would rather have it as one of the options. In such case there could be some kind of "hint warnings" during ARM compilation to suggest turning on things we consider as good to have, but have impact on pricing so we cannot turn'em on directly.

**Reply by isaacabraham** [reply]
@Dzoukr I don't believe there's any real cost implication here. It's not another service, it's just a nuget package that writes out ASP .NET logs to the local app service VM file system.
